services/web/server/src/simcore_service_webserver/projects/projects_folder_db.py

```diff
@@ -0,0 +1,87 @@
+from typing import Dict, List
+
+import sqlalchemy as sa
+from aiohttp import web
+from aiopg.sa.connection import SAConnection
+from sqlalchemy.sql import and_
+
+from ..db_models import folder_to_project
+from .projects_db import ProjectDBAPI, _convert_to_schema_names
+from .projects_exceptions import ProjectsException
+
+APP_PROJECT_FOLDER_DBAPI = __name__ + ".ProjectFolderDBAPI"
+
+
+class ProjectFolderDB(ProjectDBAPI):
+    async def add_folder_to_project(self, project):
+        print("add_folder_to_project", project, project["uuid"])
+        async with self.engine.acquire() as conn:
+            project["folder"] = await self._get_folder_by_project(
+                conn, project_uuid=project["uuid"]
+            )
+        return project
+
+    async def add_folder_to_projects(self, projects):
+        for project in projects:
+            await self.add_folder_to_project(project)
+        return projects
+
+    async def set_folder(self, user_id: int, project_uuid: str, folder_id: int) -> Dict:
+        async with self.engine.acquire() as conn:
+            project = await self._get_project(
+                conn, user_id, project_uuid, include_templates=True
+            )
+
+            folder = await self._get_folder_by_project(
```

Comment:
we have projects that are shared among different users.
What if user A puts the project in one folder and user B in one other folder? will one user lose the project or you put it in root folder if the folder does not exist?